packages/mosaic-tanstack-table-core/src/query/query-builder.ts

```diff
@@ -22,9 +22,27 @@ export function buildTableQuery<TData extends RowData, TValue>(
   const { pagination, sorting, columnFilters } = tableState;
 
   // 1. Select Columns
-  const selectColumns = mapper
-    .getSelectColumns()
-    .map((col) => mSql.column(col));
+  // FIX: Handle struct columns (e.g. "related_phrase.phrase")
+  // We iterate the mapped columns and construct the SELECT clause.
+  // If a column has a dot, we treat it as a struct access `parent.child`
+  // and ALIAS it to the original key so TanStack Table can find it flatly.
+  const selectColumns = mapper.getSelectColumns().map((col) => {
+    if (col.includes('.')) {
+      // Split "a.b" -> column("a"), column("b")
```

Comment:
I think the process of turning the dot notation into the output should be turned into its own function/util. Also, there shouldn't be so much type assertion using `as any`. Better type guards need to be used here.